### PR TITLE
Quota limit feature changes

### PR DESCRIPTION
**Pull request by shefali-malhotra**
# Description
Added details related to quota limit parameter support added in powerscale 

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/742 |

# Checklist:

- [X] Have you run a grammar and spell checks against your submission?
- [X] Have you tested the changes locally?
- [X] Have you tested whether the hyperlinks are working properly?
- [X] Did you add the examples wherever applicable?
- [ ] Have you added high-resolution images?

### CODE REVIEW

File: content/v2/csidriver/features/powerscale.md [hunk 1]
```diff
@@ -1,570 +1,659 @@
...
+      - name: pvol
+        persistentVolumeClaim:
+            claimName: isilonstaticpvc
+```
+
+5. After the pod becomes `Ready` and `Running`, you can start to use this pod and volume.
+
+## PVC Creation Feature
+
+Following yaml content can be used to create a PVC without referring any PV.
```
Comment (rsedlock1958) [inline content/v2/csidriver/features/powerscale.md:97]: Insert "The" before "Following" and remove initial uppercase from "Following"

File: content/v2/csidriver/features/powerscale.md [hunk 2]
```diff
@@ -1,570 +1,659 @@
...
+
+> For general use, update the snapshot controller YAMLs with an appropriate namespace before installing. For
+> example, on a Vanilla Kubernetes cluster, update the namespace from default to kube-system before issuing the
+> kubectl create command.
+
+### Volume Snapshot Class
+
+During the installation of CSI PowerScale driver version 2.0 and higher, no default Volume Snapshot Class will get created.
+
+Following are the manifests for the Volume Snapshot Class:
```
Comment (rsedlock1958) [inline content/v2/csidriver/features/powerscale.md:131]: Insert "The" before "Following". Remove initial uppercase on "Following"

File: content/v2/csidriver/features/powerscale.md [hunk 3]
```diff
@@ -1,570 +1,659 @@
...
+  namespace: default
+spec:
+  volumeSnapshotClassName: isilon-snapclass
+  source:
+    persistentVolumeClaimName: testvolume
+```
+
+Once the VolumeSnapshot has been successfully created by the CSI PowerScale driver, a VolumeSnapshotContent object is automatically created. Once the status of the VolumeSnapshot object has the _readyToUse_ field set to _true_ , it is available for use.
+
+Following is the relevant section of VolumeSnapshot object status:
```
Comment (rsedlock1958) [inline content/v2/csidriver/features/powerscale.md:166]: Insert "The" before "Following". Remove initial uppercase on "Following".

File: content/v2/csidriver/features/powerscale.md [hunk 4]
```diff
@@ -1,570 +1,659 @@
...
+    kind: VolumeSnapshot
+    apiGroup: snapshot.storage.k8s.io
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 5Gi
+```
+
+> Starting from CSI PowerScale driver version 2.2, it is allowed to create PersistentVolumeClaim from VolumeSnapshot with different isi paths i.e., isi paths of the new volume and the VolumeSnapshot can be different.
```
Comment (rsedlock1958) [inline content/v2/csidriver/features/powerscale.md:198]: Change sentence to read "...different isi paths can be used to create PersistentVolumeClaim from VolumeSnapshot. This means the isi paths of the new volume and the VolumeSnapshot can be different.

File: content/v2/csidriver/features/powerscale.md [hunk 5]
```diff
@@ -1,570 +1,659 @@
...
+            storage: 30Gi # Updated size from 3Gi to 30Gi
+    storageClassName: isilon-expand-sc
+```
+
+>The Kubernetes Volume Expansion feature can only be used to increase the size of a volume. It cannot be used to shrink a volume.
+
+
+## Volume Cloning Feature
+
+The CSI PowerScale driver version 1.3 and later supports volume cloning. This allows specifying existing PVCs in the _dataSource_ field to indicate a user would like to clone a Volume.
```
Comment (rsedlock1958) [inline content/v2/csidriver/features/powerscale.md:248]: Delete "The"

File: content/v2/csidriver/features/powerscale.md [hunk 6]
```diff
@@ -1,570 +1,659 @@
...
+          size: "2Gi"
+          ClusterName: "cluster1"
+```
+
+This manifest creates a pod in a given cluster and attaches a newly created ephemeral inline CSI volume to it.
+
+## Topology
+### Topology Support
+
+The CSI PowerScale driver version 1.4.0 and later supports Topology by default which forces volumes to be placed on worker nodes that have connectivity to the backend storage, as a result of which the nodes which have access to PowerScale Array are appropriately labeled. The driver leverages these labels to ensure that the driver components (controller, node) are spawned only on nodes wherein these labels exist.
```
Comment (rsedlock1958) [inline content/v2/csidriver/features/powerscale.md:347]: Delete "The". Remove comma after storage and insert a period (full stop). Replace "as a result of which the " with "This results in". Replace "are appropriately"  with "being appropriately"

File: content/v2/csidriver/features/powerscale.md [hunk 7]
```diff
@@ -1,570 +1,659 @@
...
+
+The CSI PowerScale driver version 1.4.0 and later supports Topology by default which forces volumes to be placed on worker nodes that have connectivity to the backend storage, as a result of which the nodes which have access to PowerScale Array are appropriately labeled. The driver leverages these labels to ensure that the driver components (controller, node) are spawned only on nodes wherein these labels exist. 
+  
+This covers use cases where:
+ 
+The CSI PowerScale driver may not be installed or running on some nodes where Users have chosen to restrict the nodes on accessing the PowerScale storage array. 
+
+We support CustomTopology which enables users to apply labels for nodes - "csi-isilon.dellemc.com/XX.XX.XX.XX=csi-isilon.dellemc.com" and expect the labels to be honored by the driver.
+  
+When “enableCustomTopology” is set to “true”, the CSI driver fetches custom labels “csi-isilon.dellemc.com/XX.XX.XX.XX=csi-isilon.dellemc.com” applied on worker nodes, and use them to initialize node pod with custom PowerScale FQDN/IP.
```
Comment (rsedlock1958) [inline content/v2/csidriver/features/powerscale.md:355]: Replace "use" with "uses"

File: content/v2/csidriver/features/powerscale.md [hunk 8]
```diff
@@ -1,570 +1,659 @@
...
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 5Gi
+  storageClassName: isilon
+
+```
+Note 
+- If quota limits values are specified in both storage class yaml and PVC yaml , then values mentioned in PVC yaml will get precedence.
```
Comment (rsedlock1958) [inline content/v2/csidriver/features/powerscale.md:556]: Change "limits" to "limit"

**Review by rsedlock1958 — COMMENTED** [review]
Several comments are required

File: content/v2/csidriver/features/powerscale.md [hunk 9]
```diff
@@ -1,570 +1,659 @@
...
+**Pre-Requisites:**
+
+1. Creation of secret.yaml with credentials related to one or more Clusters.
+2. Creation of (at least) one Storage class for each cluster.
+3. Creation of custom-volumesnapshot classes with proper isiPath matching corresponding storage classes.
+4. Inclusion of cluster name in volume handle, if you want to provision existing static volumes.
+
+## Consuming existing volumes with static provisioning
+
+You can use existent volumes from the PowerScale array as Persistent Volumes in your Kubernetes, perform the following steps:
```
[the remark this answers is not in the document]
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md (hunk 1 above)
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md (hunk 2 above)
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md (hunk 3 above)
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md (hunk 4 above)
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md [hunk 10]
```diff
@@ -1,570 +1,659 @@
...
+  resources:
+    requests:
+      storage: 5Gi
+```
+
+> Starting from CSI PowerScale driver version 2.2, it is allowed to create PersistentVolumeClaim from VolumeSnapshot with different isi paths i.e., isi paths of the new volume and the VolumeSnapshot can be different.
+
+## Volume Expansion
+
+The CSI PowerScale driver version 1.2 and later supports the expansion of Persistent Volumes (PVs). This expansion can be done either online (for example, when a PVC is attached to a node) or offline (for example, when a PVC is not attached to any node).
```
[the remark this answers is not in the document]
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md (hunk 5 above)
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md [hunk 11]
```diff
@@ -1,570 +1,659 @@
...
+  storageClassName: isilon
+  dataSource:
+    kind: PersistentVolumeClaim
+    name: existing-pvc
+    apiGroup: ""
+```
+
+## Controller HA
+
+The CSI PowerScale driver version 1.4.0 and later supports running multiple replicas of controller pod. At any time, only one controller pod is active(leader), and the rest are on standby.
```
[the remark this answers is not in the document]
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md (hunk 6 above)
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md (hunk 8 above)
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: Addressed the comment

File: content/v2/csidriver/features/powerscale.md (hunk 7 above)
Comment (shefali-malhotra) [reply content/v2/csidriver/features/powerscale.md]: addressed

**Review by rajkumar-palani — DISMISSED** [review]
lgtm

**Review by rajkumar-palani — APPROVED** [review]
lgtm

**Review by nitesh3108 — APPROVED** [review]

**Pull request merged**